### ISSUE TITLE
no-extra-parens | unexpected report on nested parens

### ISSUE THREAD

**Issue by seahindeniz** [eslint-stylistic/eslint-stylistic#232]
### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

TypeScript has a way of overriding a type of object with unknown.
```ts
const someKeys = Object.keys(someObject) as unknown as 'hello' | 'world';
```

In JavaScript, we could do it with the following code (I know it's irritating)
```js
const someKeys = /** @type {'hello' | 'world'} */ (/** @type {unknown} */ (
    Object.keys(someObject)
));
```

But I get the following output
```bash
/home/projects/stackblitz-starters-seafe6/index.js
  7:26  error  Unnecessary parentheses around expression  @stylistic/no-extra-parens

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.
```

However, `no-extra-parens` reports an issue even with the suggested `allowParensAfterCommentPattern: '@type'` option.

### Reproduction

https://stackblitz.com/edit/stackblitz-starters-seafe6?file=index.js

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

**Comment by seahindeniz** [comment]
There's also 
```js
const functionCall = getContext(
  /** @type {number} */ (numberOrUndefined)
);
```

**Comment by orourleyjpl** [comment]
I'm running into this same issue. It looks like it's caused by a line in the rule's `report()` function:

```
if (!isParenthesisedTwice(node)) {
```
[link](https://github.com/eslint-stylistic/eslint-stylistic/blob/ca2aabc52a9d219cd16889cc9c4ea8b04d6af054/packages/eslint-plugin/rules/no-extra-parens/no-extra-parens._js_.ts#L437)

It seems to only do the `ALLOW_PARENS_AFTER_COMMENT_PATTERN` check if there are only one set of parentheses, but in this case, there are two sets of parentheses, so it automatically is flagged without condition.

An easy fix for a PR is to remove that surrounding if statement, but someone added it there for a reason, so I don't know what other things it would break.
